cdap-watchdog-api/src/main/java/co/cask/cdap/api/metrics/MetricValue.java

```diff
@@ -16,47 +16,48 @@
 
 package co.cask.cdap.api.metrics;
 
+import co.cask.cdap.api.dataset.lib.cube.MeasureType;
+import co.cask.cdap.api.dataset.lib.cube.Measurement;
+import com.google.common.collect.ImmutableList;
+
+import java.util.Collection;
 import java.util.Map;
 
 /**
- * Carries the "raw" emitted metric data point: metric name, context, type, value, etc.
+ * Carries the "raw" emitted metric data points: context, timestamp, collection<Measurement>
  */
 public class MetricValue {
   private final Map<String, String> tags;
-  private final String name;
+  private final Collection<Measurement> metrics;
 
   /**
    * Timestamp in seconds.
    */
   private final long timestamp;
-  private final long value;
-  private final MetricType type;
+
+  public MetricValue(Map<String, String> tags, long timestamp, Collection<Measurement> metrics) {
+    this.tags = tags;
+    this.timestamp = timestamp;
+    this.metrics = metrics;
+  }
 
   public MetricValue(Map<String, String> tags, String name, long timestamp, long value, MetricType type) {
+    MeasureType measureType = type == MetricType.GAUGE ? MeasureType.GAUGE : MeasureType.COUNTER;
```

Comment:
Seems like MetricValue became exactly same as CubeFact... Maybe use it instead. Let me think - let's discuss